xdsl/dialects/x86/ops.py

```diff
@@ -1063,6 +1068,72 @@ def get_register_constraints(self) -> RegisterConstraints:
         )
 
 
+class RSSK_Operation(X86Instruction, X86CustomFormatOperation, ABC):
+    """
+    A base class for x86 AVX512 operations that have one register that is read and written to,
+    and two source registers, with masking. The z attribute enables zero masking, which sets
+    the elements of the destination register to zero where the corresponding bit in the mask
+    is zero.
+    """
+
+    T: ClassVar[VarConstraint] = VarConstraint("T", base(AVX512RegisterType))
+
+    register_in = operand_def(T)
+    register_out = result_def(T)
+    source1 = operand_def(AVX512RegisterType)
+    source2 = operand_def(AVX512RegisterType)
+    mask_reg = operand_def(AVX512MaskRegisterType)
+    z = opt_attr_def(UnitAttr)
+
+    def __init__(
+        self,
+        register_in: SSAValue[R1InvT],
+        source1: Operation | SSAValue,
+        source2: Operation | SSAValue,
+        mask_reg: Operation | SSAValue,
+        *,
+        z: bool = False,
+        comment: str | StringAttr | None = None,
+        register_out: R1InvT | None = None,
+    ):
+        if isinstance(comment, str):
+            comment = StringAttr(comment)
+
+        if register_out is None:
+            register_out = register_in.type
+
+        super().__init__(
+            operands=[register_in, source1, source2, mask_reg],
+            attributes={
+                "z": UnitAttr() if z else None,
+                "comment": comment,
+            },
+            result_types=[register_out],
+        )
+
+    def assembly_line_args(self) -> tuple[AssemblyInstructionArg | None, ...]:
+        return self.register_in, self.source1, self.source2, self.mask_reg
+
+    def assembly_line(self) -> str | None:
+        # default assembly code generator
+        instruction_name = self.assembly_instruction_name()
+        arg_str = ", ".join(
+            assembly_arg_str(arg)
+            for arg in self.assembly_line_args()
+            if arg is not None
+        )
+        if self.z:
+            arg_str += " {z}"
+        return AssemblyPrinter.assembly_line(instruction_name, arg_str, self.comment)
```

Comment:
why both of these? Does this work? If not, what's the difference?
```suggestion
    def assembly_line_args(self) -> tuple[AssemblyInstructionArg | None, ...]:
        return self.register_in, self.source1, self.source2, self.mask_reg
```

Reply:
But does it work?